### ISSUE TITLE
Add a "Photos" Category

### ISSUE THREAD

**Issue by EmperorJk13** [EmiyaSyahriel/CrossLauncher#45]
The PS3 and PSP had a Photos Category, it would be nice if the launcher had it too. Like, if the launcher had some kind of built in image viewer.

**Comment by EmiyaSyahriel** [comment]
About the Photo category, I have a bit of worry implementing this category. 

A lot of people have a tendency to *(accidentally)* stores a whole lot of images in their device, either images from camera, screenshots, meme, images from WhatsApp contacts, etc. showing the list of photos stored in device might either slows the launcher a lot, or uses too much memory.

But, I might implement it anyway, except that it will be hidden by default due to the given technical restriction, users might enable it, Only if they know these restriction.

And also I might need to refactor the code a lot to acommodate gallery streaming to reduce more memory.

<details>
  <summary>Just a personal note</summary>
   <i>I don't know how much images I have in my Tab rn since I didn't install any gallery app there, But my previous phone have 2000+ images in it even though images that actually came from it's camera is less than 100 ever since I bought it at 2018 😅</i>
</details>

**Comment by EmperorJk13** [comment]
What if the feature only added photos put in some kind of directory, for example, id.psw.vshlauncher/files/dev_flash/photos or something.

**Comment by EmperorJk13** [comment]
That way, it would not have to put every photo and use like, 2gb of ram, it would only put the photos that person would want to see.

**Comment by EmiyaSyahriel** [comment]
That's a good idea to only browse some folder for images, except that only this launcher and permitted file managers can access the `Android/data/` directory (I will do read from it anyway), You cannot easily put files in there except you have *hacked* your phone.
But by your solution, I think of these options :
- Filtering Media Database by directory list, so only photos from select directories will be added to the launcher's `Photos` list. The launcher is still need to read all images from devices media database but not all of them will make to `Photos` category, I think this will be fine although it may cause the initial `Spinning Clock` loading will be longer.
- using `File` API directly and scan the select directories from the list, This is faster than previous approach, however, Android will be upset and gives permission error if the Launcher is not `Allowed to Manage All Files`.

**Comment by EmperorJk13** [comment]
What if it had an option to scan specific directories, like if on the "Photos" category, it was empty, but it had an option to "Select a directory". That way, it could just scan the photos in _that_ specific directory. Like for example, in PPSSPP, you have to select a directory for games, or the directory for memstick.

https://github.com/EmiyaSyahriel/CrossLauncher/assets/137284132/e45c2218-0749-460b-8656-913f6ad0aba7

**Comment by EmperorJk13** [comment]
Like, it could say in the (Empty) Photos Category, "Select a directory", The person would then click on it, it would open the files manager, then select a directory, such as storage/emulated/0/Cross Launcher Images. It would then say something like, "Would you like to give Cross Launcher access to Cross Launcher Images?" Then they'd pick "Yes" and it would scan _that_ specific folder and add the images within.

**Comment by EmiyaSyahriel** [comment]
Oh yeah, the Directory Access permission request, I kinda forgot about that 😅. Sure, I'll implement it when I have the time, and It might not in the next update.

**Comment by Darthagnon** [comment]
Another idea: ability to pin photo gallery apps (e.g. [QP Gallery](https://github.com/WSTxda/QP-Gallery-Releases/releases)) under the Photos category? Similar could work for Video and music categories: pin VLC, Cloudstream, MXPlayer under the Video category, Musicolet, VLC, etc. under the Music category. 

I believe Android has some automatic categorisation built-in these days to the app drawer, maybe this could be leveraged to pin apps under the correct category?

**Comment by EmiyaSyahriel** [comment]
@Darthagnon Yeah, that's another great idea! The customization data (in addition to Android's app categorization on their manifest) is already there currently, so there shouldn't be any difficulties to implement it.
